git-clone/main.tf

```diff
@@ -26,19 +26,46 @@ variable "agent_id" {
 }
 
 locals {
-  clone_path = var.base_dir != "" ? join("/", [var.base_dir, replace(basename(var.url), ".git", "")]) : join("/", ["~", replace(basename(var.url), ".git", "")])
+  # Remove query parameters and branch name from the URL
+  url = replace(replace(var.url, "/\\?.*/", ""), "/#.*", "")
```

Comment:
Could you add an explanation of the new capability to the README? Right now it's not very discoverable in either TF desc or otherwise.

Reply:
Would it make sense to make this feature have 2 parts (initial version minimally documents (beta), and the documented with examples when "stable" :) )